### PR TITLE
SSPROD-28895: Add service principal meta data and provider_tenant_id

### PR DESCRIPTION
**Pull request by mdkulkarni15**

### CODE REVIEW

File: test/main.tf [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  component {
+    type     = "COMPONENT_SERVICE_PRINCIPAL"
+    instance = "secure-posture"
+  }
+  service_principal_metadata = jsonencode({
+    azure = {
+      active_directory_service_principal = {
+        account_enabled = true
+        display_name    = project-posture.service_principal_display_name
+        id              = project-posture.service_principal_id
```
Comment (raghavanandan) [inline test/main.tf:45]: Are these the only mandatory fields that needs to be parsed?

File: modules/services/service-principal/main.tf [hunk 2]
```diff
@@ -12,7 +12,8 @@ data "azurerm_subscription" "primary" {
 resource "azuread_service_principal" "sysdig_sp" {
   # Using Sysdig's application ID in sysdigqatenant tenant.
```
Comment (ravinadhruve10) [inline modules/services/service-principal/main.tf:13]: nit: can we remove these comments?

File: test/main.tf [hunk 3]
```diff
@@ -0,0 +1,49 @@
+provider "azurerm" {
+  features {}
+}
+terraform {
+  required_providers {
+    sysdig = {
+      source  = "sysdiglabs/sysdig"
+      version = "~> 1.15.1"
```
Comment (ravinadhruve10) [inline test/main.tf:8]: nit: whenever provider changes are ready with azure support, we'll need to update the min pinned version here.

File: test/main.tf (hunk 1 above)
Comment (ravinadhruve10) [reply test/main.tf:45]: Following up with @raghavanandan 's question - we don't need any further fields to pass in to BE for Azure SP auth?

Comment (mdkulkarni15) [reply test/main.tf]: Updated based on yesterday's testing

File: modules/services/service-principal/main.tf (hunk 2 above)
Comment (mdkulkarni15) [reply modules/services/service-principal/main.tf]: Yes, updated in last commit.

File: test/main.tf (hunk 3 above)
Comment (mdkulkarni15) [reply test/main.tf]: For now, changed this to local so that we have a sample to test locally

**Review by raghavanandan — APPROVED** [review]

**Pull request merged**